sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala

```diff
@@ -88,6 +89,16 @@ case class DescribeTableExec(
     }
   }
 
+  private def addTableConstraints(rows: ArrayBuffer[InternalRow]): Unit = {
+    if (table.constraints.nonEmpty) {
+      rows += emptyRow()
+      rows += toCatalystRow("# Constraint Information", "", "")
+      rows ++= table.constraints().map{ constraint =>
+        toCatalystRow(constraint.name(), constraint.getClass.getSimpleName, constraint.toDDL)
```

Comment:
@dengziming let's follow the output from SPIP https://docs.google.com/document/d/1EHjB4W1LjiXxsK_G7067j9pPX0y15LUF1Z5DlUPoPIo/edit?tab=t.0

```
|# Constraints               |                  |                                                   |
|pk_id                       |PRIMARY KEY       |                                                   |
|col2_not_null               |NOT NULL          |                                                   |
|col4_check                  |CHECK col4 > 10   |      
```

The second column should be the DDL